controllers/apim/authpolicy_controller.go

```diff
@@ -137,55 +136,42 @@ func (r *AuthPolicyReconciler) reconcileSpec(ctx context.Context, ap *apimv1alph
 }
 
 func (r *AuthPolicyReconciler) enforceHierarchicalConstraints(ctx context.Context, ap *apimv1alpha1.AuthPolicy) error {
-	logger, _ := logr.FromContext(ctx)
 	targetObj, err := r.fetchTargetObject(ctx, ap)
 	if err != nil {
 		return err
 	}
 
-	constrainingHosts := []string{}
-	if hr, isHR := targetObj.(*gatewayapiv1alpha2.HTTPRoute); isHR {
-		for idx := range hr.Spec.Hostnames {
-			hostname := string(hr.Spec.Hostnames[idx])
-			if len(hostname) != 0 && hostname[0] == '*' {
-				// Follows https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io%2fv1alpha2.Hostname
-				trimmedHost := strings.Trim(hostname, "*")
-				constrainingHosts = append(constrainingHosts, trimmedHost)
-			} else {
-				constrainingHosts = append(constrainingHosts, hostname)
+	netResourceHosts := make([]string, 0)
+	switch netResource := targetObj.(type) {
+	case *gatewayapiv1alpha2.HTTPRoute:
+		for _, hostname := range netResource.Spec.Hostnames {
+			netResourceHosts = append(netResourceHosts, string(hostname))
+		}
```

Comment:
If only they were of the same kind 😕

```go
netResourceHosts = append(netResourceHosts, netResource.Spec.Hostnames...)
```

Reply:
we could have our Hosts field from the rules of `[]Hostname` type, but we would need to cast anyway